### PR TITLE
LPS-140896 | master

### PR DESCRIPTION
**Pull request by ozysouza**
https://issues.liferay.com/browse/LPS-140896

cc @david-gutierrez-mesa 
cc @Tim-Cao

### CODE REVIEW

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/salesdemo/WCMSalesdemo.testcase [hunk 1]
```diff
@@ -54,9 +54,12 @@ definition {
 
 		SelectFrame.selectFrameNoLoading(locator1 = "IFrame#IFRAME_PORTLET");
 
-		AssertTextEquals.assertPartialText(
-			locator1 = "Embedded#EMBEDDED_TEXT_UPDATED",
-			value1 = "updated");
+		AssertTextEquals(
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/salesdemo/WCMSalesdemo.testcase:57]: AssertVisible(
			key_title = "Yes",
			locator1 = "Header#H1_TITLE");

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/widgetpage/WidgetPagesUseCase.testcase [hunk 2]
```diff
@@ -18,188 +24,305 @@ definition {
...
+		}
+	}
 
-		Navigator.gotoSitePage(
-			pageName = "Site Page1 Name",
-			siteName = "Site Name");
+	@description = "The user can not delete the last public page on Liferay instance."
+	@priority = "3"
+	test CannotDeleteLastPublicPageOnLiferay {
+		property testray.component.names = "Site Pages Administration";
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/widgetpage/WidgetPagesUseCase.testcase:168]: Remove duplicate property

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/widgetpage/WidgetPagesUseCase.testcase [hunk 3]
```diff
@@ -18,188 +24,305 @@ definition {
...
-			locator1 = "SitePages#LIST_GROUP_ITEM");
+			MenuItem.viewNotPresent(menuItem = "Delete");
+		}
 	}
 
+	@description = "An error is thrown when the default language has an empty friendly url in widget page."
 	@priority = "3"
-	test DefaultLanguageWithEmptyFriendlyURLThrowsErrorExceptions {
+	test DefaultLanguageWithEmptyFriendlyURLThrowsNoError {
 		property testray.component.names = "Site Pages Administration";
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/widgetpage/WidgetPagesUseCase.testcase:186]: Remove duplicate property

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/widgetpage/WidgetPagesUseCase.testcase [hunk 4]
```diff
@@ -210,688 +333,639 @@ definition {
...
-
-		User.firstLoginPG(
-			password = "test",
-			userEmailAddress = "test@liferay.com");
 	}
 
+	@description = "User admin can choose when a private widget page will be visible to assigned role."
 	@priority = "5"
-	test SitesPrivatePagesPermissions {
+	test PrivatePageVisibleToAssignedRole {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/widgetpage/WidgetPagesUseCase.testcase:340]: CannotViewPrivatePageAsUserWithoutPermissions

**Review by Tim-Cao — COMMENTED** [review]
Hi @ozysouza, Could you update changes according to comments? Thanks

**Comment by ozysouza** [comment]
> Hi @ozysouza, Could you update changes according to comments? Thanks

Hi @Tim-Cao, PR updated following your comments, could you take a look at? Thanks!

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/widgetpage/WidgetPagesUseCase.testcase [hunk 5]
```diff
@@ -18,875 +24,942 @@ definition {
...
-
-		Navigator.gotoNavTab(navTab = "SEO");
-
-		Page.editPageMetadata(
-			HTMLTitle = "S.P.A.C.E.Home",
-			metaTagsDescription = "Welcome to S.P.A.C.E., a graduate-level school for those passionate about expanding their horizons through science, research, and space travel",
-			metaTagsKeywords = "space, education, space travel, graduateschool");
-
-		Navigator.gotoPage(pageName = "Site Page Edit");
+	test CannotDeleteLastPublicPageOnLiferay {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/widgetpage/WidgetPagesUseCase.testcase:101]: CannotDeleteLastPublicPageOnDefaultSite

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/widgetpage/WidgetPagesUseCase.testcase [hunk 6]
```diff
@@ -18,875 +24,942 @@ definition {
...
-		Blogs.addEntryWithPermissions(
-			entryContent = "Blogs Entry Content",
-			entryTitle = "Blogs Entry Title GMB2",
-			viewableBy = "Site Members");
+			ControlMenu.viewHeaderTitle(headerTitle = "Test Page Name");
 
-		Navigator.gotoPage(pageName = "Blog and Message Board Page");
+			AssertElementNotPresent(
+				locator1 = "Home#ERROR_MESSAGE_ALERT_ERROR",
+				value1 = "Not Found");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/widgetpage/WidgetPagesUseCase.testcase:524]:
```suggestion
		AssertTextNotPresent(value1 = "The requested resource could not be found");
```

**Review by Tim-Cao — COMMENTED** [review]
Hi @ozysouza, Could you update according to comments? Thanks

**Comment by ozysouza** [comment]
> Hi @ozysouza, Could you update according to comments? Thanks

Hey @Tim-Cao, PR updated with the last changes you requested. :D

**Comment by Tim-Cao** [comment]
Good job! @ozysouza, see changes https://github.com/liferay-echo/liferay-portal/compare/c230bf9dd9bafa34aabad53ef34c22db55d2584c...c637a679eb1bdad5ff14ceadb8e6c51595f4d4e8

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**